src/main/java/net/snowflake/client/jdbc/cloud/storage/SnowflakeAzureClient.java

```diff
@@ -934,6 +933,17 @@ private static URI buildAzureStorageEndpointURI(String storageEndPoint, String s
     return storageEndpoint;
   }
 
+  /*
+   * getUserDefinedMetadata
+   * Method introduced to avoid inconsistencies in custom headers handling, since these are defined on drivers side
+   * e.g. some drivers might internally convert headers to canonical form.
+   */
+  private Map<String, String> getUserDefinedMetadata(CloudBlob blob) {
```

Comment:
Maybe we can make it consistent for all types of clouds? One helper function to rule them all?

Reply:
let's also add some tests to protect it in all cloud providers